### ISSUE TITLE
Just Register: Deploy-FabricApplication and Publish-NewServiceFabricApplication

### ISSUE THREAD

**Issue by cmenzi** [Azure/service-fabric-issues#488]
Hello 

I try to just register new application type with the `Deploy-FabricApplication -DeployOnly and OverwriteBehavior VetoUpgrade`, which calls `Publish-NewServiceFabricApplication` with `Action = Register`.

The problem is that Publish-NewServiceFabricApplication checks if there already an application exists.

```
#Line 167
$app = Get-ServiceFabricApplication -ApplicationName $ApplicationName
```

I think is only necessary when I also want to create an application, but I only like to register it.

In the `ApplicationParameters/Cloud.xml` file there is the `Name` specified, but this should, in my opinion, only be used when I also create the application. 

What I want to achieve is just register the application type which, basically is Connect, Copy, Register and I like to use the `Deploy-FabricApplication.ps1`

The scripts behaves correctly:

- if no application is created
- the first time when I use RegisterAndCreate, second time with just Register it fails.
- I change `Name` in `Cloud.xml` to something where no application exists.

Is there a parameter I've forgot?

**Comment by MikkelHegn** [comment]
`VetoUpgrade` is an OverrideUpgradeBehavior, not OverwriteBehavior.

There are three scenarios where this is possible:
1. No applications of that application type are already running in the cluster.
2. If you set the OverwriteBehavior to SameAppTypeAndVersion, running applications of that app type and version will be removed before registering the new app type version.
3. If you set the OverwriteBehavior to Always, running applications of that app type will be removed before registering the new app type version.

**Comment by cmenzi** [comment]
Yes, I used this to get $IsUpdate = $false in Deploy-FabricApplication.ps1

For there one scenario missing: Just "Register" an new version of a given applicationtype, I don't want any application to be removed or upgraded. 

My scenario: 

1.  Register an given ApplicationType with new version -> Make the ApplicationType available to create an application from it
2. Upgrade the new ApplicationType for selective Applications (Testing)
3. Upgrade every remaining Application with the "good" ApplicationType (Version)

**Comment by MikkelHegn** [comment]
That scenario is not supported today, I'll log a bug to look in to this. Not sure we can make it generic enough to be parts of those scripts.
The Deploy-FabricApplication.ps1 is meant as yours to customize for needs like these. Copying the Publish-NewServiceFabricApplication script in to your projects script folder and customize to your scenario would probably be a more suitable solution.

**Comment by cmenzi** [comment]
Thanks

I've found a way to which does the trick:

```
#Line 254 Deploy-FabricApplication.ps1
else
{
    $Action = "RegisterAndCreate"
    if ($DeployOnly)
    {
        $Action = "Register"
        # This is workaround, until Publish-NewServiceFabricApplication does not check application names in case if just Register.
        $ApplicationName = Get-ApplicationNameFromApplicationParameterFile $publishProfile.ApplicationParameterFile
        $PublishParameters['ApplicationName'] = "$ApplicationName/$([Guid]::NewGuid())";
    }

    $PublishParameters['Action'] = $Action
    $PublishParameters['OverwriteBehavior'] = $OverwriteBehavior
    $PublishParameters['SkipPackageValidation'] = $SkipPackageValidation
    Write-Host "Publish-NewServiceFabricApplication"
    Publish-NewServiceFabricApplication @PublishParameters
}
```
Just using a unique ApplicationName, which results in just register.

**Comment by MikkelHegn** [comment]
@amanbha
